src/object.d

```diff
@@ -10,13 +10,14 @@
 
 module object;
 
-private
+version(D_BetterC)
 {
-    extern (C) Object _d_newclass(const TypeInfo_Class ci);
-    extern (C) void rt_finalize(void *data, bool det=true);
+    nothrow:
```

Comment:
This doesn't work (attributes in a `version` block don't apply to the outer scope).

Reply:
Thanks.  What's the proper idiom, then?

Reply:
I don think that there is one. I tried something along the lines of : 

```d
import std.stdio: writeln;

version(V1)
{
    enum versionl = true;
}
else
{
}

static if (versionl)
{
    notrow:
}

void f() 
{
    writeln("bla");
}
```

But it doesn't work since it looks like `nothrow:` cannot be placed inside a `static if` body. If static if doesn't work and `nothrow:` inside a version body does not work for the outer scope I see no other solution then to actually declare the code as a string and mixin it in each branch. I hope that someone has a better idea, cause the one I proposed is awful.

Reply:
You could put the optionally nothrow code into a template mixin, and mixin the template on each leaf of the version statement.

Reply:
> You could put the optionally nothrow code into a template mixin, and mixin the template on each leaf of the version statement.

Oh, man, that would be great if it worked.  Please illustrate with an example, as I don't quite follow  You can see my prior attempts here:  https://forum.dlang.org/post/uhgzgmowqcczczrdtsol@forum.dlang.org